prog/dftb+/lib_solvation/born.F90

```diff
@@ -200,6 +222,16 @@ subroutine initialize(self, input, nAtom, species0, speciesNames, latVecs)
     self%param = input%TGBParameters
     self%rho(:) = input%vdwRad(:) * input%descreening(:)
 
+    if (allocated(self%sasaCont) .and. allocated(input%hBondPar)) then
+      if (any(input%hBondPar /= 0.0_dp)) then
+        allocate(self%hBondStrength(nAtom))
+        do iAt1 = 1, nAtom
+          iSp1 = species0(iAt1)
+          self%hBondStrength(iAt1) = input%hBondPar(iSp1) / self%sasaCont%probeRad(iSp1)**2
```

Comment:
Might want to consider a future PR to pre-compute the inverse of the probeRad data. Unless the compiler is very clever it won't work out that there are only a few of these and it can save (x~25) flops with a multiplication by the stored inverses.

Reply:
This is actually storing the inverses in `hBondStrength` for efficiency reasons. I see no need to optimize this part of the constructor, except for memory usage of the `hBondStrength` array, but this version makes it more convenient to use later (also think about a more empirical model for the GBSA where this parameter is indeed atom resolved by some function of the geometry / topology).

Reply:
Was actually thinking of this _specific_ implementation, so something like
invRad = 1.0_dp / self%sasaCont%probeRad
do iAt1 = 1, nAtom
as nAtom >> nSpecies most of the time and nSpecies < ~10. But this isn't time or memory critical here, so as you wish.

Reply:
It is still a step in the constructor, so I don't see the point of requiring the allocation of a scratch array. Maybe one could simplify a bit by a temporary scalar with
```suggestion
          invRad = 1.0_dp / self%sasaCont%probeRad(iSp1)
          self%hBondStrength(iAt1) = input%hBondPar(iSp1) * invRad * invRad
```

Reply:
The scratch array is tiny and goes out of scope rapidly (and is probably what a sufficiently smart compiler will effectively do). Not seeing the proposed change as either simpler or faster, so happy to stay with original.